qt/scientific_interfaces/Muon/ALCDataLoadingView.cpp

```diff
@@ -76,6 +72,24 @@ void ALCDataLoadingView::initialize() {
   m_ui.alpha->setValidator(new QDoubleValidator(0, 99999, 10, this));
 
   m_ui.runs->doButtonOpt(MantidQt::API::FileFinderWidget::ButtonOpts::None);
+
+  // Set up connections
+  connect(m_ui.help, SIGNAL(clicked()), this, SLOT(help()));
+  connect(m_ui.load, SIGNAL(clicked()), this, SLOT(notifyLoadClicked()));
+  connect(m_ui.instrument, SIGNAL(currentTextChanged(QString)), this, SLOT(instrumentChanged(QString)));
+  connect(m_ui.runs, SIGNAL(fileTextChanged(const QString &)), this, SLOT(notifyRunsEditingChanged()));
+  connect(m_ui.runs, SIGNAL(findingFiles()), this, SLOT(notifyRunsEditingFinished()));
+  connect(m_ui.runs, SIGNAL(fileFindingFinished()), this, SLOT(notifyRunsFoundFinished()));
+  connect(m_ui.manageDirectoriesButton, SIGNAL(clicked()), this, SLOT(openManageDirectories()));
+  connect(m_ui.periodInfo, SIGNAL(clicked()), this, SLOT(notifyPeriodInfoClicked()));
+  connect(m_ui.runsAutoAdd, SIGNAL(toggled(bool)), this, SLOT(runsAutoAddToggled(bool)));
+
+  // Watcher to check if directory to load data from changes
+  // Need to connect using function pointers because directoryChanged() is somehow not recognised as a signal
+  connect(m_watcher, &QFileSystemWatcher::directoryChanged, [this]() { m_presenter->setDirectoryChanged(true); });
+
+  // Timer to send timeout() signal intermitently
+  connect(m_timer, SIGNAL(timeout()), this, SLOT(notifyTimerEvent()));
```

Comment:
There is a new signal-slot syntax which is now preferred in C++ (although its barely used in Mantid). The new syntax allows you to get compile-time warnings if there is a slot or signal that is missing. Previously, this would go undetected unless you keep an eye on the command prompt when opening your Qt interface
https://wiki.qt.io/New_Signal_Slot_Syntax

Reply:
Ah, so that's the difference! I encountered this style before but because the syntax looked a bit clunky to me at the time I just assumed that it was an older version of making connections. If I had looked this up it would have saved me some time during development :laughing: